### PR TITLE
Improved cleaning of URL to prevent validation errors

### PR DESCRIPTION
**Pull request by mikewalch**

### CODE REVIEW

**Comment by keith-turner** [comment]
Build will fail until fluo-io/fluo-recipes#36 is merged

File: modules/core/src/main/java/io/fluo/webindex/core/DataUtil.java [hunk 1]
```diff
@@ -22,6 +22,9 @@
 
 public class DataUtil {
 
+  public static String URL_SEP_REGEX = "[\\/\\?\\#]";
```
Comment (keith-turner) [inline modules/core/src/main/java/io/fluo/webindex/core/DataUtil.java]: Was not sure escaping was needed in regex char set.  Also I was not sure if would honor escaping or treat `\` as a character in the set.  I wrote the following program to see.

```
    String re1 = "[\\/\\?\\#]";
    String re2 = "[/?#]";

    String data = "one/two?three#four\\five";

    System.out.println(Arrays.asList(data.split(re1)));
    System.out.println(Arrays.asList(data.split(re2)));
```

It printed 

```
[one, two, three, four\five]
[one, two, three, four\five]
```

Comment (mikewalch) [inline modules/core/src/main/java/io/fluo/webindex/core/DataUtil.java]: I don't remember why I thought the escapes were necessary.  Removed in e577e45ff

File: modules/data/src/test/java/io/fluo/webindex/data/util/LinkUtilTest.java [hunk 2]
```diff
@@ -28,6 +28,12 @@ public void testBasic() throws Exception {
     Assert.assertTrue(LinkUtil.isValid("http://1.2.3.4/"));
     Assert.assertTrue(LinkUtil.isValid("http://a.b.c.d.com/1/2/3/4/5"));
     Assert.assertTrue(LinkUtil.isValid("http://a.b.com:281/1/2"));
+    Assert.assertTrue(LinkUtil.isValid("http://A.B.Com:281/a/b"));
+    Assert.assertTrue(LinkUtil.isValid("http://A.b.Com:281/A/b"));
+    Assert.assertTrue(LinkUtil.isValid("http://a.B.Com?A/b/C"));
+    Assert.assertTrue(LinkUtil.isValid("http://A.Be.COM"));
+    Assert.assertTrue(LinkUtil.isValid("http://a.com:/test"));
+    Assert.assertTrue(LinkUtil.isValid("http://z.com:"));
```
Comment (keith-turner) [inline modules/data/src/test/java/io/fluo/webindex/data/util/LinkUtilTest.java]: In addition to calling isValid.  Should test also check expected output of cleanUrl?  

```java
Assert.assertEquals("http://z.com", cleanUrl("http://z.com:"));
```

File: modules/core/src/main/java/io/fluo/webindex/core/DataUtil.java [hunk 3]
```diff
@@ -106,14 +109,36 @@ public static String toUri(URL url) {
   }
 
   public static String cleanUrl(String url) {
-    String cleanUrl = url.trim();
-    if (cleanUrl.length() >= 8) {
-      if (cleanUrl.substring(0, 7).equalsIgnoreCase("http://")) {
-        cleanUrl = "http://" + cleanUrl.substring(7);
-      } else if (cleanUrl.substring(0, 8).equalsIgnoreCase("https://")) {
-        cleanUrl = "https://" + cleanUrl.substring(8);
+    StringBuilder cleanUrl = new StringBuilder();
```
Comment (keith-turner) [inline modules/core/src/main/java/io/fluo/webindex/core/DataUtil.java]: Is the following the only cleaning this method does?

 * Strips `:` when nothing follows it in hostname

Comment (mikewalch) [inline modules/core/src/main/java/io/fluo/webindex/core/DataUtil.java]: It also trims whitespace and makes the host part of the url lowercase

File: modules/data/src/test/java/io/fluo/webindex/data/util/LinkUtilTest.java (hunk 2 above)
Comment (mikewalch) [inline modules/data/src/test/java/io/fluo/webindex/data/util/LinkUtilTest.java]: Added to DataUtilTest in 151fcb2e304e9

**Comment by keith-turner** [comment]
+1

**Pull request merged**